zipline/pipeline/engine.py

```diff
@@ -12,6 +12,7 @@
     with_metaclass,
 )
 from numpy import array
+from odo.utils import copydoc
```

Comment:
`odo` is an optional requirement for Zipline.  We should just copy this function and put it somewhere like `zipline.utils.sharedoc`.